xarray/core/dataset.py

```diff
@@ -3867,6 +3867,79 @@ def differentiate(self, coord, edge_order=1, datetime_unit=None):
                 variables[k] = v
         return self._replace_vars_and_dims(variables)
 
+    def integrate(self, dim, datetime_unit=None):
+        """ integrate the array with the trapezoidal rule.
+
+        .. note::
+            This feature is limited to simple cartesian geometry, i.e. coord
+            must be one dimensional.
+
+        Parameters
+        ----------
+        dim: str, or a sequence of str
+            Coordinate(s) used for the integration.
+        datetime_unit
+            Can be specify the unit if datetime coordinate is used. One of
+            {'Y', 'M', 'W', 'D', 'h', 'm', 's', 'ms', 'us', 'ns', 'ps', 'fs',
+             'as'}
+
+        Returns
+        -------
+        integrated: Dataset
+
+        See also
+        --------
+        DataArray.integrate
+        numpy.trapz: corresponding numpy function
+        """
+        if not isinstance(dim, (list, tuple)):
+            dim = (dim, )
+        result = self
+        for d in dim:
+            result = result._integrate_one(d, datetime_unit=datetime_unit)
+        return result
+
+    def _integrate_one(self, dim, datetime_unit=None):
+        from .variable import Variable
+
+        if dim not in self.variables and dim not in self.dims:
+            raise ValueError('Coordinate {} does not exist.'.format(dim))
```

Comment:
I think splitting these checks into two would be a little clearer:
- "cannot integrate over dimension {} because it does not exist" (for `dim not in self.dims`)
- "cannot integrate over dimension {} because there is no corresponding coordinate" (for `dim not in self.variables`)